source/extensions/filters/http/cache/cache_header_utility.cc

```diff
@@ -0,0 +1,109 @@
+#include "extensions/filters/http/cache/cache_header_utility.h"
+
+#include <vector>
+
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/header_utility.h"
+#include "common/http/headers.h"
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+namespace {
+const std::string bytes_{"bytes"};
+}
+
+std::vector<RawByteRange> CacheHeaderUtility::getRanges(const Http::HeaderMap& request_headers) {
+  // Range headers are only valid on GET requests so make sure we don't get here
+  // with another type of request.
+  ASSERT(request_headers.Method() != nullptr &&
+         request_headers.Method()->value() == Http::Headers::get().MethodValues.Get);
```

Comment:
This ASSERT would be better as two asserts, so any failures would make it obvious which condition failed.